src/main/java/com/google/sps/data/NonPersistentUserRepository.java

```diff
@@ -0,0 +1,61 @@
+package com.google.sps.data;
+
+import java.lang.StringBuilder;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * User repository that stores users in an ArrayList so that we don't need to access
+ * datastore for testing.
+ */
+ public class NonPersistentUserRepository implements UserRepository {
+     private Collection<User> allUsers;
+
+     public NonPersistentUserRepository() {
+         allUsers = new ArrayList<User>();
+     }
+
+     public void addFakeMentors() {
+        User mentorA = new User("Andre");
+        mentorA.addProductArea("Machine Learning");
+        mentorA.addProductArea("DoS");
+
+        User mentorB = new User("Jerry");
+        mentorB.addProductArea("Electrical Engineering");
+        mentorB.addProductArea("DoS");
+
+        User mentorC = new User("Julie");
+        mentorC.addProductArea("Machine Learning");
+        mentorC.addProductArea("Security");
+
+        addUser(mentorA);
+        addUser(mentorB);
+        addUser(mentorC);
+     }
+
+     public void addUser(User user) {
+         if(!allUsers.contains(user)) {
```

Comment:
If you switch allUsers to a set, adding the same user twice should automatically collapse into a single entry. It will also make removeUser and addUser O(log(n)) if you use Set, or O(1) if you use unordered set / HashSet (HashSet is generally prefered, unless you want to be able to, say, compare two sets by coiterating them, then you'd want a datastructure like Set whose iteration order is fixed)

Reply:
We discussed & I'm not switching it because the order matters for some tests :)